### ISSUE TITLE
Secondary bank changes introduces error in nu_scatter scores

### ISSUE THREAD

**Issue by nelsonag** [mit-crpg/openmc#460]
the nu_scatter tallies, which rely upon the difference between p % wgt and p % last_wgt to inform how the neutrons produced in (n,xn) reactions (i.e., p % wgt / p % last_wgt = x), have not been updated to reflect the fact that secondary particles are now explicitly tracked.

**Comment by wbinventor** [comment]
I have been wondering about the nu-scatter tallies. I am seeing a bias when
usong my multi-group cross-sections from OpenMC in OpenMOC, and I think it
is because nu-scatter is not picking up the nu. I wanted to dig into this a
bit further before posting an issue. @nelsonag are you under the same
impression as I that nu is not being captured correctly?

**Comment by nelsonag** [comment]
Yep, I'm seeing a new bias between my mg mc code and the new omc tallies.
That's how I've stumbled upon this.

**Comment by wbinventor** [comment]
Very interesting. I see a -200 pcm bias between MC and MOC in a simple
infinite medium for a variety of energy group structures. I previously
confirmed that this is approx the reactivity effect coming from Zr
scattering, though I wanted to ensure that it isn't a separate issue with
my data processing code first. I have confirmed that the nu-scatter and
scatter (analog) tally results are bitwise identical for this particular
problem (1,2,8,40,70 groups) even with 100 million histories. This leads me
to believe that scattering production is no longer being picked up by the
nu-scatter score.

**Comment by nelsonag** [comment]
Ironically enough this morning I was also running an infinite homogeneous problem too and am seeing a 208 pcm bias.

Looking at the source code, p % wgt no longer is scaled to capture the new particles (as is correct when they are explicitly tracked) but the tallying code still assumes p % wgt contains the multiplication.

**Comment by wbinventor** [comment]
I'm glad to hear that I'm not the only crazy one here :-) Cool, perhaps @paulromano may have a good idea of how to quickly fix this issue.

**Comment by nelsonag** [comment]
Yeah, whew! Me too :-)

I think the best way would be to multiply p % last_wgt by the multiplicity at tally time. I just don't know at the moment if all the information is available to the general tally routine or if it needs to be reworked to have access to all it needs.

**Comment by nelsonag** [comment]
I think I've got it fixed, testing now.

**Comment by paulromano** [comment]
I think we need to be careful about this one. I see the same thing, i.e. that scatter and nu-scatter give the exact same result, but this also seems to be case for v0.7.0 where they are barely different. Moreover, the scatter score dropped going from v0.7.0 to now. Here are some results for examples/pincell:

Version 0.7.0
- scatter (tracklength) = 29.6848 +/- 2.29e-2
- scatter (analog) = 29.6938 +/- 2.30e-2
- nu-scatter (analog) = 29.6954 +/- 2.30e-2

develop
- scatter (tracklength) = 29.5867 +/- 2.41e-2
- scatter (analog) = 29.5950 +/- 2.39e-2
- nu-scatter (analog) = 29.5950 +/- 2.39e-2

It seems that both scores are messed up. I need to cogitate on this one a bit more...

**Comment by wbinventor** [comment]
The scatter and nu-scatter scores may be "barely different" in v0.7 but
they are different enough to resolve a -200pcm bias when using MGXS
computed with OpenMC for LWR analysis in a deterministic code. The fact
that both scores are different since the inclusion of the secondary
particle bank is certainly interesting.

**Comment by paulromano** [comment]
@wbinventor I'm willing to bet that if you used just normal scatter scores (instead of nu-scatter) from v0.7.0 in your MGXS, the bias would be much less than 200pcm. Can you try that experiment for me?

**Comment by nelsonag** [comment]
@paulromano was that change with the same energy group structure in the examples/pincell problem (i.e., 2-group)?  That difference in v0.7 you point out looks reasonable for a 2-group difference.  I have similar order for a 1-group pincell problem and that was also worth approximately 200pcm.

**Comment by nelsonag** [comment]
Though either way, both scatter and nu-scatter have gone down since v0.7 and that definitely needs some looking in to.  if I get any ideas I'll let you know

**Comment by wbinventor** [comment]
Sure, one can compute the analytical eigenvalue for homogeneous infinite medium with one group 'total', 'nu-fission' and 'nu-scatter' tallies simply as k = 'nu-fission' / ('total' - 'nu-scatter'). This is effectively how the eigenvalue is computed in OpenMC (in practice we use the ratio of successive fission sources but it converges to the same result). 

I ran a simple infinite homogeneous medium with these tallies (analog only) for three different version of OpenMC. You can see the XML input files I used at the bottom of this post. The following tabulates the results for this one group calculation using v0.6.2, v0.7.0 and the latest develop version of the code:

| Version   | OpenMC keff            | scatter                             | keff        | nu-scatter                     | keff        |
| ------------ |:--------------------------: | --------------------------------: | ----------: | -------------------------------: | ----------: |
| v0.6.4      | 1.15927 +/- 0.00031 | 2.5748e+01 +/- 5.09e-03 | 1.15728 | 2.5750e+01 +/- 5.1e-03 | 1.15929 |
| v0.7.0      | 1.15927 +/- 0.00031 | 2.5748e+01 +/- 5.09e-03 | 1.15728 | 2.5750e+01 +/- 5.09e-03 | 1.15929 |
| develop   | 1.15885 +/- 0.00031 | 2.5745e+01 +/- 4.74e-03 | 1.15694 | 2.5745e+01 +/- 4.74e-03 | 1.15694 |

As you can see, a seemingly minor (<1%) difference between 'nu-scatter' and 'scatter' translates into a bias of a few hundred pcm (for the LWR case). This same bias is confirmed for multiple energy group structures (2,8,16,25,40,70), using OpenMOC in place of the analytical solution. In this case I used NNDC cross-sections.

**geometry.xml**
```xml
<?xml version='1.0' encoding='utf-8'?>
<geometry>
    <cell id="1" material="10000" name="cell" surfaces="10000 -10001 10002 -10003" universe="0" />
    <surface boundary="reflective" coeffs="-0.63" id="10000" type="x-plane" />
    <surface boundary="reflective" coeffs="0.63" id="10001" type="x-plane" />
    <surface boundary="reflective" coeffs="-0.63" id="10002" type="y-plane" />
    <surface boundary="reflective" coeffs="0.63" id="10003" type="y-plane" />
</geometry>
```

**materials.xml**
```xml
<?xml version='1.0' encoding='utf-8'?>
<materials>
    <default_xs>71c</default_xs>
    <material id="10000" name="moderator">
        <density units="g/cc" value="5.0" />
        <nuclide ao="0.006886019" name="U-238" />
        <nuclide ao="0.01450188" name="O-16" />
        <nuclide ao="0.002116053" name="Zr-90" />
        <nuclide ao="0.028999667" name="H-1" />
        <nuclide ao="0.000114142" name="U-235" />
    </material>
</materials>
```
**tallies.xml**
```xml
<?xml version='1.0' encoding='utf-8'?>
<tallies>
    <tally id="1">
        <scores>flux total nu-fission absorption scatter nu-scatter</scores>
        <estimator>analog</estimator>
    </tally>
</tallies>
```

**settings.xml**
```xml
<?xml version='1.0' encoding='utf-8'?>
<settings>
    <eigenvalue>
        <particles>100000</particles>
        <batches>100</batches>
        <inactive>5</inactive>
    </eigenvalue>
    <source>
        <space type="box">
            <parameters>-0.63 -0.63 -0.63 0.63 0.63 0.63</parameters>
        </space>
    </source>
</settings>
```

**Comment by paulromano** [comment]
Thanks for that @wbinventor. It looks like the differences I was seeing between the scatter scores were actually just due to not running enough neutrons. When I ran more, the v0.7.0 and develop scatter reaction rates agree, but nu-scatter doesn't. So fortunately it looks like the problem is only with nu-scatter (and I think @nelsonag's fix is correct).

**Comment by paulromano** [comment]
Closed by #461
